pdns/dnssecinfra.cc

```diff
@@ -168,13 +168,20 @@ std::unique_ptr<DNSCryptoKeyEngine> DNSCryptoKeyEngine::makeFromISCString(DNSKEY
   return dpk;
 }
 
+auto DNSCryptoKeyEngine::makeFromPEMFile(DNSKEYRecordContent& drc, const std::string& filename, std::FILE* fp, const uint8_t algorithm) -> std::unique_ptr<DNSCryptoKeyEngine>
```

Comment:
Is there any benefit to the form you are using with regard to the more "usual" one?
```suggestion
std::unique_ptr<DNSCryptoKeyEngine> DNSCryptoKeyEngine::makeFromPEMFile(DNSKEYRecordContent& drc, const std::string& filename, std::FILE* fp, const uint8_t algorithm)
```

Reply:
It's modern C++ at this point, and suggested by clang-tidy. IIRC the reason for it is due to an ambiguity with the old-style when combined with templates in certain cases.

I think we should just systematically move to the new-style, what do you think?

Reply:
Personally I'm not convinced by the trailing return type syntax, except of course when it is actually needed. I'm also a bit worried that moving everything to the new style would require a lot of changes and make backporting fixes more painful. That being said I'm not strongly opposed to it, especially if there is some benefits that I'm not seeing yet :)

Reply:
Had a short discussion with @Habbie over lunch about this and we'll stick with the old-style for non-template functions. I'll also see if I can disable that particular diagnostic from clang-tidy.